main/acle.md

```diff
@@ -4932,6 +5044,1381 @@ assertions, which may speed up access to aligned data (and will fault
+
+*   if F2 is a [shared-ZA](#shared-za) function, F1 shares its ZA state
+    with F2.
+
+*   otherwise, F1's ZA state is unchanged by F2.
+
+Again the analogy is with passing or not passing a `pseudo_za_state`
+reference to F2.
+
+Functions that have ZA state can use the [SME intrinsic functions](#sme-intrinsic-functions)
+to manipulate that state. These intrinsic functions are themselves
+shared-ZA functions and so share ZA state with their callers.
+
+### `asm` statements and ZA
+
+Some ACLE implementations might support the GNU “inline asm” extension.
+For implementations that do, suppose that an inline asm occurs in a
+function F. There are then two cases:
+
+1.  If F has [ZA state](#za-state), PSTATE.ZA is guaranteed to be 1 on entry
+    to the inline asm. The inline asm must finish with PSTATE.ZA equal
+    to 1, otherwise the behavior is undefined.
+
+    The inline asm can indicate that it reads the current contents of ZA
+    and/or that it changes the contents of ZA by adding `"za"` to the
+    asm's clobber list. Using the clobber list for this purpose is a
+    syntactic convenience: it does not fit the normal semantics for clobbers.
+
+    If the inline asm does not have a `"za"` clobber but nevertheless
+    reads the current contents of ZA or changes the contents of ZA,
+    the behavior is undefined.
+
+2.  If F does not have [ZA state](#za-state), the inline asm must
+    “comply with the lazy saving scheme”, in the sense of
+    [[AAPCS64]](#AAPCS64). The behavior in other cases is undefined.
+
+    The inline asm is [ill-formed](#ill-formed) if it has a `"za"` clobber.
+
+## SME attributes
+
+All of the attributes described in this section can be specified
+using the GNU `__attribute__` syntax. Their names can be used
+directly or with two underscores added to each side. For example:
+
+``` c
+  __attribute__((arm_streaming))
+  __attribute__((__arm_streaming__))
+```
+
+Specifying an attribute multiple times is equivalent to specifying it once.
+
+Some of the attributes described in this section apply to function types.
+Their semantics are as follows:
+
+*   If the attribute is attached to a function declaration or definition,
+    it applies to the type of the function.
+
+*   If the attribute is attached to a function type, it applies to that type.
+
+*   If the attribute is attached to a pointer to a function type, it applies
+    to that function type.
+
+*   Otherwise, the attribute is [ill-formed](#ill-formed).
+
+Except where noted otherwise, function types that have an attribute are
+incompatible with function types that do not. For example:
+
+``` c
+  #define ATTR __attribute__((arm_streaming))
+
+  typedef void (*n_callback_type)(void);
+  n_callback_type n_callback_ptr;
+  void n_extern_function(void);
+  void n_local_function(void) { ... }
+
+  typedef ATTR void (*s_callback_type)(void);
+  s_callback_type s_callback_ptr;
+  ATTR void s_extern_function(void);
+  ATTR void s_local_function(void) { ... }
```

Comment:
I suspect that there is an implicit meaning in the use of the `s_` and `n_` prefixes, but I have hard times finding it. Would it be possible to explain it in the text, even if just as comment in the code example?